File: frontend/src/api/quiz.ts
import api from "./axiosInstance";

export const generateQuiz = async (no_of_questions: number, file_name: string) => {
    try {
        const response = await api.post("/quiz/generateQuiz", {
            no_of_questions: no_of_questions,
            file_name: file_name
        });
        return response.data;
    } catch (error) {
        console.error("Error generating quiz:", error);
        throw error;
    }
}

File: backend/app/api/routes/quiz.py
from fastapi import APIRouter
from app.config import vector_store_define
from app.data.models.Quiz import QuizRequest
from app.services.quizGenerate import generate_quiz

router = APIRouter()

@router.post("/generateQuiz")
async def get_quiz(quiz_request: QuizRequest):

    vector_store = vector_store_define()

    retrieved_docs = vector_store.similarity_search(
        "what are the main facts ?", 
        k=quiz_request.no_of_questions + 3,
        filter={"filename": quiz_request.file_name}
    )

    chunks = [doc.page_content for doc in retrieved_docs] # extract text content from Langchain Document format that accepts from generate_quiz

    quiz = generate_quiz(chunks, no_of_questions=quiz_request.no_of_questions)

    return quiz